contribute/design/1.x/apis/cstor_volume.md

```diff
@@ -0,0 +1,359 @@
+# Move `CStorVolumes`, `CStorVolumeReplica` resources Schema to version `v1`
+
+## Table of Contents
+
+* [Introduction](#introduction)
+* [Goals](#goals)
+* [Non-Goals](#non-goals)
+* [Schema Proposal](#volume-schema-proposal)
+* [Risk and Mitigations](#risks-and-mitigations)
+* [CRD Migration and Graduation](#graduation-criteria)
+
+## Introduction
+
+This proposal highlights the enhancements and migration of current `CStorVolumes` and `CStorVolumeReplicas` schema and
+proposes improvements.
+
+# Goals
+
+The major goal of this document is to freeze the APIs schema for `CStorVolumes` and `CStorVolumeReplica`.
+Apart from this the document focuses on following aspects too:
+
+- Migrating the `CStorVolumes` and `CStorVolumeReplica` CRD to `v1` apiversion.
+- Identify the breaking and upgrade changes and documentation around that.
+- List of possible high level status fields on `CStorVolumes` and `CStorVolumeReplica`.
+- Enhancements for operations based resource events based in `CStorVolumes` and `CStorVolumeReplicas`
+
+# Enhancements
+
+- Generate conditional events for different operations
+- Volume Replica Rebuild status information for individual replicas
+- VolumeReplica capacity details using logical referenced values
+- Enhanced and updated `CStorVolumeReplicas` health state on different failures observed via controllers
+- Enhanced and updated `CStorVolumes` health state on different failures observed via controllers
+- Add BlockSize and Compression properties to configure them on per volume bases
+
+# Non-Goals
+
+- Introduction of new fields in the `CStorVolumes` and `CStorVolumeReplica` to deliver feature etc.
+- List of possible events that should be recorded on `CStorVolumes` and `CStorVolumesReplicas`.
+- List of conditions that should be put on the `CStorVolumes` and `CStorVolumesReplicas` CRs.
+
+# Volume Schema Proposal
+
+## `CStorVolumes` APIs
+
+```go
+// +genclient
+// +k8s:openapi-gen=true
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CStorVolume describes a cstor volume resource created as custom resource
+type CStorVolume struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              CStorVolumeSpec   `json:"spec"`
+	VersionDetails    VersionDetails    `json:"versionDetails"`
+	Status            CStorVolumeStatus `json:"status"`
+}
+
+// CStorVolumeSpec is the spec for a CStorVolume resource
+type CStorVolumeSpec struct {
+	// Capacity represents the desired size of the underlying volume.
+	Capacity          resource.Quantity `json:"capacity"`
+	TargetIP          string            `json:"targetIP"`
```

Comment:
Add one line comments to each of these items. While most of the iSCSI fields are understood, what does `NodeBase` mean?

Reply:
Its a just a constant name used to construct the `iqn` name, used just as info, not required to be in API, i have removed it and declare as a constant in code itself.